packages/api/src/pins.js

```diff
@@ -261,85 +263,176 @@ export async function pinGet (request, env, ctx) {
   }
 
   /** @type { ServiceApiPinStatus } */
-  const response = {
-    requestId: pinRequest._id,
-    created: pinRequest.created,
-    // TODO populate delegates
-    delegates: [],
-    status: getPinningAPIStatus(pinRequest.pins),
-    pin: {
-      cid: pinRequest.requestedCid,
-      name: pinRequest.name,
-      // TODO populate origins and meta
-      origins: [],
-      meta: {}
-    }
-  }
-
-  return new JSONResponse(response)
+  const pin = getPinStatus(pinRequest)
+  return new JSONResponse(pin)
 }
 
 /**
+ * List all the pins matching optional filters.
+ * When no filter is specified, only successful pins are returned.
+ *
  * @param {import('./user').AuthenticatedRequest} request
  * @param {import('./env').Env} env
  * @param {import('./index').Ctx} ctx
  */
 export async function pinsGet (request, env, ctx) {
-  const { cid, name, match, status } = request.params
+  const url = new URL(request.url)
+  const urlParams = new URLSearchParams(url.search)
+  const params = Object.fromEntries(urlParams)
+
+  const result = parseSearchParams(params)
+  if (result.error) {
+    return new JSONResponse(result.error, { status: 400 })
+  }
+
+  let pinRequests
+  const opts = result.data
+
+  try {
+    pinRequests = await env.db.listPAPinRequests(request.auth.authToken._id, opts)
+  } catch (e) {
+    console.error(e)
+    return notFound()
+  }
+
+  const pins = pinRequests.map((pinRequest) => getPinStatus(pinRequest))
+  return new JSONResponse({
+    count: pins.length,
```

Comment:
This should be the total number of PinRequests that satisfy the query rather than the ones returned.

ie. if there are 100 Pinrequests for a auth token, limit is 10,  `count` should be 10 but the array of `results` just contains 10 objects

Reply:
Good catch. Now corrected.